json/index.md

```diff
@@ -1,4 +1,25 @@
----
----
+# Goals of chapter
+I want to show a scale model of how you could JSON decoding (and encoding) in Roc. Doing so will demonstrate using ADTs for modelling your data, pattern matching, function reuse and how to build a small Codec libary.
```

Comment:
Minor terminology note for the actual text of the chapter - in all the Roc documentation we talk about "tag unions" rather than referring to them as ADTs (OCaml programmers might point out that they're more like polymorphic variants anyway), so it'd be good to keep consistent with that terminology!

Reply:
okay sure - so does it make sense to say a tag union is a sum of tag types? where each tag can be a value by itself or it can have a payload associated with it?

Reply:
I'd say that's accurate, although in my experience anyone who's familiar with the term "sum types" understands them right away and it doesn't matter what explanation I use, so I try to focus on people who are used to languages that don't have any form of the feature. 😄 

I like to talk about it in terms of "alternatives" - like "this type could be one of several alternatives..."

Reply:
right gotcha that makes sense!